projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.component.ts

```diff
@@ -676,4 +766,70 @@ export class IgxExcelStyleSearchComponent implements AfterViewInit, OnDestroy {
             this.searchValue = this.searchInput.value;
         }
     }
+
+    private onArrowUpKeyDown() {
+        if (this.focusedItem && this.focusedItem.index === 0 && this.virtDir.state.startIndex === 0) {
+            this.searchInput.focus();
+            this.onFocusOut();
+        } else {
+            this.navigateItem(this.focusedItem ? this.focusedItem.index - 1 : 0);
+        }
+        this.setActiveDescendant();
+    }
+
+    private onArrowDownKeyDown() {
+        const lastIndex = this.virtDir.igxForOf.length - 1;
+        if (this.focusedItem && this.focusedItem.index === lastIndex) {
+            this.cancelButton.nativeElement.focus();
+            this.onFocusOut();
+        } else {
+            this.navigateItem(this.focusedItem ? this.focusedItem.index + 1 : 0);
+        }
+        this.setActiveDescendant();
+    }
+
+    private onHomeKeyDown() {
+        this.navigateItem(0);
+        this.setActiveDescendant();
+    }
+
+    private onEndKeyDown() {
+        this.navigateItem(this.virtDir.igxForOf.length - 1);
+        this.setActiveDescendant();
+    }
+
+    private onActionKeyDown() {
+        const dataItem = this.displayedListData[this.focusedItem.index];
+        const args: IChangeCheckboxEventArgs = {
+            checked: !dataItem.isSelected,
+            checkbox: this.checkboxes.find(x => x.value === dataItem)
+        }
+        this.onCheckboxChange(args);
+    }
+
+    private navigateItem(index: number) {
+        if (index === -1 || index >= this.virtDir.igxForOf.length) {
+            return;
+        }
+        const direction = index > (this.focusedItem ? this.focusedItem.index : -1) ? Navigate.Down : Navigate.Up;
+        const scrollRequired = this.isIndexOutOfBounds(index, direction);
+        this.focusedItem = {
+           id: this.getItemId(index),
+           index: index,
+           checked: this.virtDir.igxForOf[index].isSelected
+        };
+        if (scrollRequired) {
+            this.virtDir.scrollTo(index);
+        }
+    }
+
+    private isIndexOutOfBounds(index: number, direction: Navigate) {
```

Comment:
I think we already have a similar public method on the igx-for-of directive, that we could probably re-use: https://github.com/IgniteUI/igniteui-angular/blob/16.1.x/projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts#L763

Reply:
I've logged a separate task for refactoring some of the navigation services here?
I think we can make this as part of it as well

Reply:
Sure.